### PR TITLE
CSPL-1923: Addition of code and unittests to download the app packages

### PR DESCRIPTION
**Pull request by kumarajeet**
**Adding functionality to download app packages from Azure blob**
Changes related to CSPL-1923:
Extending  `azureblobclient.go` to allow downloading app packages from azure blob containers (buckets) 
Extending the unit tests to tests the download functionality as well as enhance the app lists unit tests

### CODE REVIEW

**Review by jryb — APPROVED** [review]

**Review by vivekr-splunk — APPROVED** [review]

File: pkg/splunk/client/azureblobclient_test.go [hunk 1]
```diff
@@ -167,230 +298,210 @@ func TestAzureBlobGetAppsList(t *testing.T) {
...
+	wantRequest, _ := http.NewRequest("GET", "https://mystorageaccount.blob.core.windows.net/appscontainer1/adminAppsRepo/app1.tgz", nil)
+	respdata := "This is a test body of an app1.tgz package. In real use it would be a binary file but for test it is just a string data"
 
-	var vol enterpriseApi.VolumeSpec
-	var err error
+	mclient.AddHandler(wantRequest, 200, respdata, nil)
 
+	// Get App source and volume from spec
 	appSource := appFrameworkRef.AppSources[0]
+	vol, _ := GetAppSrcVolume(ctx, appSource, &appFrameworkRef)
```
Comment (gaurav-splunk) [inline pkg/splunk/client/azureblobclient_test.go:452]: why are we not checking the error here?

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -16,5 +16,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: splunk/splunk-operator
-  newTag: 1.1.0
\ No newline at end of file
+  newName: controller
```
Comment (gaurav-splunk) [inline config/manager/kustomization.yaml:19]: are you sure this is correct? CC: @vivekr-splunk ?

Comment (kumarajeet) [reply config/manager/kustomization.yaml]: I think I pushed local change, so I reverted that to what we had originally.

File: config/manager/kustomization.yaml [hunk 3]
```diff
@@ -16,5 +16,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: controller
-  newTag: latest
+  newName: splunk/splunk-operator
+  newTag: 1.1.0
```
Comment (gaurav-splunk) [inline config/manager/kustomization.yaml:20]: Also, is this tag also correct? it should be something 2.x, isn't it? @vivekr-splunk ?

Comment (vivekr-splunk) [reply config/manager/kustomization.yaml:20]: yes it should be 2.0.0 but creation of image is done dynamically, just for the purpose of having right number we can change this to 2.0.0 . pipeline always uses right tag name when it creates

File: config/manager/kustomization.yaml (hunk 2 above)
Comment (vivekr-splunk) [reply config/manager/kustomization.yaml]: no changes to config directory is required. unless we are doing changes to CRD.

File: config/manager/kustomization.yaml (hunk 3 above)
Comment (kumarajeet) [reply config/manager/kustomization.yaml]: Thanks Vivek and Gaurav. Updated to 2.0.0

**Review by gaurav-splunk — APPROVED** [review]

**Pull request merged**